server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.sqlcheck.rule;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.springframework.jdbc.core.JdbcOperations;
+
+import com.oceanbase.odc.core.shared.constant.DialectType;
+import com.oceanbase.tools.sqlparser.statement.Statement;
+import com.oceanbase.tools.sqlparser.statement.delete.Delete;
+import com.oceanbase.tools.sqlparser.statement.insert.Insert;
+import com.oceanbase.tools.sqlparser.statement.update.Update;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @description:{@link OracleAffectedRowsExceedLimit}
+ * 
+ * @author: zijia.cj
+ * @date: 2024/10/24 13:06
+ * @since: 4.3.3
+ */
+@Slf4j
+public class OracleAffectedRowsExceedLimit extends BaseAffectedRowsExceedLimit {
+    public static final String ODC_TEMP_EXPLAIN_STATEMENT_ID = "ODC_TEMP_EXPLAIN_STATEMENT_ID";
+
+    private final JdbcOperations jdbcOperations;
+    private final DialectType dialectType;
+
+    public OracleAffectedRowsExceedLimit(@NonNull Long maxSqlAffectedRows, DialectType dialectType,
+            JdbcOperations jdbcOperations) {
+        super(maxSqlAffectedRows);
+        this.jdbcOperations = jdbcOperations;
+        this.dialectType = dialectType;
+    }
+
+    /**
+     * Get supported database types
+     */
+    @Override
+    public List<DialectType> getSupportsDialectTypes() {
+        return Arrays.asList(DialectType.OB_ORACLE, DialectType.ORACLE);
+    }
+
+    /**
+     * Base method implemented by Oracle types
+     */
+    @Override
+    public long getStatementAffectedRows(Statement statement) {
+        long affectedRows = 0;
+        if (statement instanceof Update || statement instanceof Delete || statement instanceof Insert) {
+            String originalSql = statement.getText();
+            try {
+                if (this.jdbcOperations == null) {
+                    throw new IllegalStateException("JdbcOperations is null, please check your connection");
+                } else {
```

Comment:
`else`  seems that it can be removed, and the code will be better understood.

Reply:
ok